### ISSUE TITLE
Get intralink information from rustdoc

### ISSUE THREAD

**Issue by orium** [orium/cargo-rdme#177]

**Comment by RReverser** [comment]
Trying out the branch - looks pretty good so far, but one thing I'm finding missing is ability to pass custom Cargo args (maybe as `cargo rdme -- ...[cargo options]...`?) as some projects require `--features` to be passed in to compile at all.

**Comment by orium** [comment]
I'm glad you gave it a try. There's still some work to do (not to mention the code is really messy right now), but I should be able to work on it occasionally.

I was thinking of starting simply with `--all-features`. I might add a first class `features` option (both in the command line and config file) in the future that can overrides `--all-features`.

Not sure about arbitrary cargo flags, at least for now. I can be convinced otherwise if there are compelling use cases.

**Comment by RReverser** [comment]
> I was thinking of starting simply with `--all-features`.

I think that might be problematic, as a lot of repos tend to declare features like `nightly` or `no_std` or `unstable` etc, which are usually not desirable and might break more things than `--all-features` fixes.

> Not sure about arbitrary cargo flags, at least for now. I can be convinced otherwise if there are compelling use cases.

Yeah I can't really think of other compelling cases for now, but it is a pattern that other cargo-based tools tend to follow, so figured might be easiest to support. Explicit `--features` sounds good though.

**Comment by orium** [comment]
> I think that might be problematic, as a lot of repos tend to declare features like nightly or no_std or unstable etc, which are usually not desirable and might break more things than --all-features fixes.

That's true. I'll have an explicit features options and default to whatever default features the package has.

**Comment by angrynode** [comment]
This branch is great! Some notes:

- it allows unqualified links as asked in #151, removing the cargo doc warning about unnecessary complete path
- it does not allow to resolve links without an actual URI (so you have to write `[Api](Api)` instead of just `[Api]` in usual rustdoc)
- i had to bump some crates and `EXPECTED_RUSTDOC_FORMAT_VERSION` to 42 but all went fine
- it complained at runtime that nightly toolchain was not installed... i wonder if there could be a way to statically bundle whatever's needed from nightly in a single cargo-rdme binary

**Comment by RReverser** [comment]
Just wanted to check how rustdoc-based implementation is going, since it's been a while now. Did it just fall off the radar priority-wise or have there been any blockers?

**Comment by orium** [comment]
No blockers. I just need to put some time aside to finish it. I intend to finish that this month (but no promises).

**Comment by orium** [comment]
> The features will be configurable via the configuration file and command line options.

Implemented in PR #236.
